generatebundlefile/bundle.go

```diff
@@ -81,42 +84,76 @@ func (projects Project) NewPackageFromInput() (*api.BundlePackage, error) {
 }
 
 // AddMetadata adds the corresponding metadata to the crd files.
-func AddMetadata(s api.PackageBundleSpec) BundleGenerate {
-	return BundleGenerate{
+func AddMetadata(s api.PackageBundleSpec, name string) api.PackageBundle {
+	bundleName := name
+	return api.PackageBundle{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       api.PackageBundleKind,
 			APIVersion: SchemeBuilder.GroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      s.KubeVersion,
+			Name:      bundleName,
 			Namespace: bundle.ActiveBundleNamespace,
 		},
 		Spec: s,
 	}
 }
 
+func CheckSignature(bundle *api.PackageBundle) (bool, error) {
+	annotations := bundle.ObjectMeta.Annotations
+	if annotations != nil {
+		return true, nil
+	}
+	return false, nil
+}
+
+func AddSignature(bundle *api.PackageBundle, signature string) (*api.PackageBundle, error) {
+	annotations := map[string]string{}
+	if signature == "" || bundle == nil {
+		return nil, fmt.Errorf("Error adding signature to bundle, empty signature, or bundle entry\n")
+	}
+	annotations = map[string]string{
+		"eksa.aws.com/signature": signature,
+	}
+	bundle.ObjectMeta.Annotations = annotations
+	return bundle, nil
+}
+
+func CompareSignature(bundle *api.PackageBundle, signature string) (string, error) {
+	annotations := map[string]string{}
+	if signature == "" || bundle == nil {
+		return "", fmt.Errorf("Error adding signature to bundle, empty signature, or bundle entry\n")
+	}
+	annotations = map[string]string{
+		"eksa.aws.com/signature": signature,
+	}
+	if annotations["eksa.aws.com/signature"] == bundle.ObjectMeta.Annotations["eksa.aws.com/signature"] {
+		return signature, nil
+	}
+	return bundle.ObjectMeta.Annotations["eksa.aws.com/signature"], nil
+}
+
 // MarshalBundleSpec will create yaml objects from bundlespecs
-func (bundleSpec BundleGenerate) MarshalBundleSpec() ([]byte, error) {
-	marshallables := []interface{}{bundleSpec}
+func MarshalBundleSpec(bundle api.PackageBundle) ([]byte, error) {
+	marshallables := []interface{}{bundle}
 	resources := make([][]byte, len(marshallables))
 	for _, marshallable := range marshallables {
 		resource, err := yaml.Marshal(marshallable)
 		if err != nil {
-			return nil, fmt.Errorf("failed marshalling resource for bundle spec: %v", err)
+			return nil, fmt.Errorf("failed marshalling resource for bundle: %v", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("marshalling resource for bundle: %w", err)
```
The `%w` wraps the error, so upstream users can identify specific errors.